### DISCUSSION TITLE
can application made using spin  run anywhere ?

### DISCUSSION THREAD

**Discussion by Horiodino** [spinframework/spin#1849] [General]
i was planning to build some project using web-assembly and why not to use spin .but can we run it anywhere ? i have seen the docks and i see a comparison of spin and lambda .just confused that's all. anywhere in the sense outside fermyon cloud like what if i want to deliver my application in my local server .

**Comment by itowlson** [comment]
Yes, you can run Spin on your own Linux, Mac or Windows server.  (Not BSD at the moment, unfortunately.)  It is absolutely _not_ bound to Fermyon Cloud.  The `spin up` command runs your application on the included HTTP server (or Redis listener, or whatever), and you can configure things like ports and TLS certificates through the command line or environment variables.

To be clear, `spin up` by itself _just_ runs the application.  If your use case requires failure recovery, load balancing, or all those other good things, and you're using the Spin CLI, then you'll need to set those up separately.  If you're interested in more robust, scalable systems, here are a couple of talks from people who have built those around Spin outside of Fermyon Cloud:

* [Thorsten Hans](https://youtu.be/ai5p5oubTm8?t=699)
* [PJ Laszkowicz](https://youtu.be/-Xz3tyxm-RU?t=713)

Please let us know if you need more information!

**Reply by Horiodino** [reply]
thank you :smiley:
